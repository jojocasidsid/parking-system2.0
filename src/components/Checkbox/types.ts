import { Control } from 'react-hook-form'

export interface IProps {
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	control?: Control<any, object> // To do: change any type
	label: string
	name: string
	defaultValue?: boolean
	disabled?: boolean
	error?: string
	fullWidth?: boolean
	required?: boolean
	noController?: boolean
	value?: boolean
	onChange?: () => {}
}
