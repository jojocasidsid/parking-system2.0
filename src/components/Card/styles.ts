import styled from '@emotion/styled'

export const CardWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlotHeader = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`
