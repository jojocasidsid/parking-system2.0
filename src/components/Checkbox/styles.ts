import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	padding: 18px;
`
