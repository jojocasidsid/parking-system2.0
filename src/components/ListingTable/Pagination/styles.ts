import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
