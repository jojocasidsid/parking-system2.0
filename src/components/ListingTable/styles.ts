import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export const StyledTypography = styled(Typography)`
	color: #30364380;
	font-weight: bold;
	font-size: 14px;
`
