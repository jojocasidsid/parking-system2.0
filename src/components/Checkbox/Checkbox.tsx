import React from 'react'
import { FormControlLabel } from '@mui/material'
import { Controller } from 'react-hook-form'
import { IProps } from './types'
import { StyledCheckbox, StyledRoot } from './styles'

const Checkbox = ({
	name,
	control,
	label,
	defaultValue,

	...rest
}: IProps) => (
	<FormControlLabel
		label={label}
		control={
			<Controller
				name={name}
				control={control}
				defaultValue={!!defaultValue}
				render={({ field }) => (
					<StyledRoot>
						<StyledCheckbox
							checked={field.value}
							onChange={field.onChange}
							{...rest}
						/>
					</StyledRoot>
				)}
			/>
		}
	/>
)

export default Checkbox
