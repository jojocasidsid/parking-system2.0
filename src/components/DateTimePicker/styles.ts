import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface ITheme {}
interface IInput extends ITheme {
	errors?: boolean
	theme?: Theme
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, errors }: IInput) => {
		if (errors) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: 8px 16px 8px 16px;
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
		border-color: ${({ theme, errors }: IInput) =>
			errors
				? theme?.palette.error.main
				: theme?.palette.text.secondary} !important;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 8px;
	}

	.MuiFormHelperText-root {
		color: ${({ theme }) => theme.palette.error.main};
		margin-left: 0;
	}
`
