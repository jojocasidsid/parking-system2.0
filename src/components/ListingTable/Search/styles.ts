import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
