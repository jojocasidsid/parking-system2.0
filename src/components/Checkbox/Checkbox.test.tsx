import React from 'react'

// packages
import * as yup from 'yup'
import { ThemeProvider } from '@mui/material/styles'
import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import renderer from 'react-test-renderer'

// config
import theme from 'config/theme'

// components
import Checkbox from './Checkbox'

export const validationSchema = yup.object({
	test: yup.boolean(),
})
export interface IValidationSchema
	extends yup.InferType<typeof validationSchema> {}
export const defaultValues: IValidationSchema = {
	test: true,
}

describe('Checkbox component', () => {
	it('should create snapshot correctly', () => {
		const Component = () => {
			const { control } = useForm({
				mode: 'onTouched',
				defaultValues,
				resolver: yupResolver(validationSchema),
			})

			return (
				<ThemeProvider theme={theme}>
					<Checkbox control={control} label='Test' name='test' />
				</ThemeProvider>
			)
		}

		const tree = renderer.create(<Component />)

		expect(tree).toMatchSnapshot()
	})
})
