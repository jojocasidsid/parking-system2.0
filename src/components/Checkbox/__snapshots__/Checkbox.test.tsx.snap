// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should create snapshot correctly 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
>
  <div
    className="css-bjmv2m"
  >
    <span
      className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-1q3ox7e-MuiButtonBase-root-MuiCheckbox-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <input
        checked={true}
        className="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
  </div>
  <span
    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1che66z-MuiTypography-root"
  >
    Test
  </span>
</label>
`;
